File: backend/models/ai.py
from pydantic import BaseModel, Field

from config import MAX_JOB_DESCRIPTION_CHARS
from models.document import Change


class Suggestion(BaseModel):
    id: str
    paragraph_id: str
    start: int
    end: int
    original_text: str
    alternatives: list[str]
    reason: str


class AnalyzeRequest(BaseModel):
    doc_id: str
    # Bounded: this goes straight into the prompt, so an unbounded value costs
    # the caller tokens and costs the server request time.
    job_description: str = Field(max_length=MAX_JOB_DESCRIPTION_CHARS)
    changes: list[Change] = Field(default_factory=list)


class AnalyzeResponse(BaseModel):
    suggestions: list[Suggestion]

File: frontend/src/types/ai.ts
export interface Suggestion {
    id: string;
    paragraph_id: string;
    start: number;
    end: number;
    original_text: string;
    alternatives: string[];
    reason: string;
    state: 'open' | 'accepted';
    applied_text?: string;
}

export interface AnalyzeResult {
    suggestions: Omit<Suggestion, 'state' | 'applied_text'>[];
}
